### DISCUSSION TITLE
How to convert Bson filter from org.bson.conversions.Bson to morphia Filter

### DISCUSSION THREAD

**Discussion by omar1413** [MorphiaOrg/morphia#2399] [General]
Is there is a way to convert bson filter to Morphia Filter

    Morphia Version: 2.2.10

**Comment by evanchooly** [comment]
There is not.  The two APIs are similar but have some variations here and there.  It shouldn't be too hard to convert, though.  In many cases, it could be as simple as changing the import statements if memory serves.

**Comment by omar1413** [comment]
Can I utilize the same technique explained in this context [1823](https://github.com/MorphiaOrg/morphia/issues/1823) ?

**Reply by evanchooly** [reply]
Anything that's in that `Document` will eventually get handled by a `Codec` so passing in a `Bson` as part of that `Document` *should* work, in theory.  I'm not a fan of the `Bson` API (hence the "mirrored" API) but if that works for you, by all means go for it.  You might find it simpler to work with the more common Java/JDK types, however.
